Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Generator.cs

```diff
@@ -212,11 +277,89 @@ public void Execute(GeneratorExecutionContext context)
             }
         }
 
+        private static ExecutableAssembly GenerateExecutableAssemblySource(
+            GeneratorExecutionContext context,
+            DiagnosticReporter diagnosticReporter,
+            SyntaxReceiver receiver,
+            List<LambdaFunctionModel> lambdaModels)
+        {
+            // Check 'Amazon.Lambda.RuntimeSupport' is referenced
+            if (context.Compilation.ReferencedAssemblyNames.FirstOrDefault(x => x.Name == "Amazon.Lambda.RuntimeSupport") == null)
+            {
+                diagnosticReporter.Report(
+                    Diagnostic.Create(
+                        DiagnosticDescriptors.MissingDependencies,
+                        Location.None,
+                        "Amazon.Lambda.RuntimeSupport"));
+                
+                return null;
+            }
+
+            foreach (var methodDeclaration in receiver.MethodDeclarations)
+            {
+                var model = context.Compilation.GetSemanticModel(methodDeclaration.SyntaxTree);
+                var symbol = model.GetDeclaredSymbol(methodDeclaration) as IMethodSymbol;
+
+                // Check to see if a static main method exists in the same namespace that has 0 or 1 parameters
+                if (symbol.Name != "Main" || !symbol.IsStatic || symbol.ContainingNamespace.Name != lambdaModels[0].LambdaMethod.ContainingAssembly || (symbol.Parameters.Length > 1)) 
+                    continue;
+                
+                diagnosticReporter.Report(
+                    Diagnostic.Create(
+                        DiagnosticDescriptors.MainMethodExists,
+                        Location.None));
+                    
+                return null;
+            }
+
+            return new ExecutableAssembly(
+                lambdaModels,
+                lambdaModels[0].LambdaMethod.ContainingNamespace);
```

Comment:
I suspect this is the code that is causing the bug that if there are no Lambda functions we give back the "AWSLambda001 This is a bug. Please run the build ..." compilation error.

Reply:
Added a specific Diagnostic error to handle this.